with-openai/config.yaml

```diff
@@ -0,0 +1,4 @@
+configurationset:
+  - configuration:
+      name: openai_config
+      openai_auth: Bearer STEPZEN_OPENAI_API_KEY
```

Comment:
Would rename to `OPENAI_API_KEY` to prevent confusion with the stepzen key

Reply:
If we want to use an env var in `config.yaml` it has to start with `STEPZEN_` - the CLI only substitutes such vars.
If starting with `STEPZEN_` is a problem, then we should consider dropping / changing this CLI feature.